projects/ruby/fuzz_ruby_gems.c

```diff
@@ -155,6 +155,9 @@ int run_fuzz_function(struct ByteStream *bs, struct TargetFunction *fcn) {
       goto out;
     }
     args[i] = v;
+    // Make sure that `args[i]` doesn't get garbage collected until after the
+    // call.
+    rb_gc_register_address(&args[i]);
```

Comment:
This probably works, but isn't ideal as these methods are intended to be called extremely rarely, like once at boot (`rb_gc_unregister_address` scans through a large linked list.)

Some alternatives:
* We could store `VALUE *args` on the stack instead of heap, so the conservative collector sees them. `VALUE args[MAX_NARGS]`
* Use a Ruby array instead of a `VALUE *` pointer for storing the args
* Wrap `TargetCall` as a Ruby data structure and provide a mark function to mark the args (I like this option ther least)

Reply:
The stack option seems the simplest. I'll do that.